### PR TITLE
Post

### PR DESCRIPTION
**Pull request by timktchan**
User can create post & view all posts

Fix indentation
Post-client.js > set up static.js for userpost.html on client side
Post.js > Prev: For created post, "username" shos objectId. Fixed: name will be shown on "username"
HTML > change on index.html & post.html

### CODE REVIEW

File: public/application.js [hunk 1]
```diff
@@ -19,10 +17,150 @@ $(document).ready(function(){
...
+
+  $('#signInSubmit').click(function(){
+    event.preventDefault();
+    console.log('hello signin');
+    var username = $('.signIn > input[id="siusername"]');
+    var password = $('.signIn > input[id="sipassword"]');
+
+    account.signIn(username,password);
+    
+    // $.ajax({
```
Comment (f3r) [inline public/application.js:87]: no commented code pls

File: public/application.js [hunk 2]
```diff
@@ -19,10 +17,150 @@ $(document).ready(function(){
...
+
+  $('#signOut').click(function(){
+    event.preventDefault();
+    console.log('hello signout');
+
+    account.signOut();
   });
 
+});
+  // $('#signUpSubmit').click(function(){
```
Comment (f3r) [inline public/application.js]: NO COMMENTED CODE PLEASE

File: public/post-client.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    event.preventDefault();
+      post.title    = $('#addTitle').val();
+      post.content  = $('#addContent').val();
+      post.category = $('#addCategory').val();
+      post.photo    = $('#addPhoto').val();
+
+    post.addPost();
+  });
+});
+  // var getAllPosts = function(){
```
Comment (f3r) [inline public/post-client.js]: NO COMMENTED CODE PLEASE

File: routes/posts.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        })
+      }
+    },
+    // GET one post
+    {
+      method: 'GET',
+      path: '/posts/{id}',
+      handler: function(request, reply){
+        var post_id  = encodeURIComponent(request.params.id);
+        // .id retrieve from path , path from window's browser url?
```
Comment (f3r) [inline routes/posts.js]: you can't retrieve from window browser url, because this code executes in the backend

File: routes/posts.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  ]);
+next();
+}
+
+exports.register.attributes = {
+  name: 'posts-routes',
+  version: '0.0.1'
+}
+    // {
+    //   // Retrieve all posts by a specific user
```
Comment (f3r) [inline routes/posts.js:92]: NO COMMENTED CODE PLEASE

File: routes/sessions.js [hunk 6]
```diff
@@ -49,18 +49,29 @@ exports.register = function(server, options, next) {
...
-        });
+        var session = request.session.get('allaboutgrowth_session');
+        //authenticare through cookie to make sure u're deleting yr own session
+        Auth.authenticated(request, function(result){
+          if (!session) { 
+            return reply({ "message": "Already logged out" });
+          }
+          db.collection('sessions').remove({ "session_id": session.session_id }, function(err, writeResult) {
+            if (err) { return reply('Internal MongoDB error', err); }
+            // whats writeresult
```
Comment (f3r) [inline routes/sessions.js]: http://api.mongodb.org/java/current/com/mongodb/WriteResult.html

File: routes/users.js [hunk 7]
```diff
@@ -10,17 +9,16 @@ exports.register = function(server, options, next){
...
-          var user = {
-            username: request.payload.user.username,
-            email:    request.payload.user.email,
-            password: request.payload.user.password,
-            bio: request.payload.user.bio,
-            profilepic: request.payload.user.profilepic
-          };
-
+          var user = request.payload.user;
+          // var user = {
```
Comment (f3r) [inline routes/users.js]: NO COMMENTED CODE PLEASE

File: templates/index.html [hunk 8]
```diff
@@ -4,8 +4,10 @@
   <title>AllAboutGrowth</title>
   <script type="text/javascript" src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
   <script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>
-
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
```
Comment (f3r) [inline templates/index.html]: load `jquery` before `bootstrap`

File: templates/posts.html [hunk 9]
```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>AllAboutGrowth</title>
+  <script type="text/javascript" src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
+  <script type="text/javascript" src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
```
Comment (f3r) [inline templates/posts.html]: `jquery` included TWICE and two different versions

**Comment by f3r** [comment]
Please fix issues before you can merge, thanks

**Comment by timktchan** [comment]
Remove 90% of comment code, remaining are those I may use for other features

**Pull request merged**